### ISSUE TITLE
Vertical text being displayed on on Yomichan overlay with new windows 10 computer.

### ISSUE THREAD

**Issue by sailikk** [FooSoft/yomichan#178]
![image](https://user-images.githubusercontent.com/46624965/62839247-a18b1700-bc55-11e9-9057-d40f03911056.png)

I don't know what's causing this but it started happening when I switched my old Windows 10 computer for a new one. I'm not sure what's changed but this is somewhat annoying. I have the English JMEdict installed, just like I had before...

**Comment by siikamiika** [comment]
Can you share some information about your environment like the Yomichan version, font and default styles?
![2019-08-12_20-33-14](https://user-images.githubusercontent.com/5602483/62885113-b2648700-bd40-11e9-8dc4-3f379e6226cb.png)
It could be a weird sum of those, but either way, in a recent testing version of Yomichan you can set custom CSS to your popup after enabling advanced options. I'm not sure if the build [here](https://foosoft.net/projects/yomichan/index.html#installation) has it, but a new version is scheduled to release soon.

**Comment by sailikk** [comment]
I'm using chrome, so I don't know if how you're seeing the font tab.. firefox? But I mean, we can see from CSS that the font is supposed to be "sans-serif". 

I do not know how to check the version of Yomichan, but I downloaded it just a few days ago on this computer, so unless a new patch has come out, this should be the latest, maybe. Not sure what you mean by default styles, does the CSS answer your question? I'm not the best at web stuff unfortunately...

![image](https://user-images.githubusercontent.com/46624965/62984494-6decf000-be01-11e9-8a45-dfbb49e9b561.png)

**Comment by toasted-nutbread** [comment]
@sailikk There have been several styling updates since this issue was created. Does Yomichan still have this issue for you? Looking to resolve this if it's still a problem, or close otherwise.

**Comment by sailikk** [comment]
@toasted-nutbread I am using Yomichan v20.1.12.0, so the latest release, still on Windows computer, Chrome version 79.0.3945.130

The issue persists.

![image](https://user-images.githubusercontent.com/46624965/73116149-9411d280-3eff-11ea-83f1-bc96463c163d.png)

**Comment by toasted-nutbread** [comment]
So after a bit of testing, it looks like the font you're seeing is [Meiryo](https://docs.microsoft.com/en-us/typography/font-list/meiryo). I didn't have it installed by default, so I had to install some of Window's Japanese language packs to get it to show up.

I can get my popup to look similar, but it isn't exhibiting any issue yet.
![image](https://user-images.githubusercontent.com/11037431/73116879-c0cbe700-3f0b-11ea-9079-46c605a69fc8.png)

However, my scaling doesn't exactly match what I'm seeing in your image. What's the resolution of your screen, do you have any system-level DPI scaling, and what was the zoom level you have that site at? I had to zoom in to 125% to get it close to yours, but my scale was still bigger, so I'm trying to get it to match as close as possible.

**Comment by sailikk** [comment]
My Windows system level DPI is set to 120%. I never thought that that would be causing it. Indeed, setting it to 125% does get rid of it, but my current setting works better for certain applications.

**Comment by siikamiika** [comment]
Interesting, I wonder if this is a bug in Windows font rendering. I don't remember seeing 120% as a default option in the drop down in Windows 10 last time I tested. What are you using to set the DPI scaling?

**Comment by toasted-nutbread** [comment]
> My Windows system level DPI is set to 120%.

I had to use the "Advanced scaling settings" window to get a scaling of 120%; did you also have to use that window?

My layout matches quite closely now, but I'm still not seeing the issue.
![test](https://user-images.githubusercontent.com/11037431/73125893-34a4d880-3f7a-11ea-8b2a-0e0ea1312611.png)

Comparison:
![test](https://user-images.githubusercontent.com/11037431/73125895-38385f80-3f7a-11ea-895d-6fdf7eaa3936.gif)

There are two main differences I see:
* It looks like the ClearType font rendering is different on our systems (different colors on the antialiasing). I don't think this should affect anything, but I'll mess with my settings.
* There's an additional orange highlight on the selected text in your image. Do you know what that is coming from?

**Comment by toasted-nutbread** [comment]
Actually, I just reproduced the issue. I turned off ClearType font rendering to get it occur, so I'll continue debugging from there.

![image](https://user-images.githubusercontent.com/11037431/73125996-4a66cd80-3f7b-11ea-8889-7aeef5cc4538.png)

**Comment by toasted-nutbread** [comment]
Here is something else interesting: check out the text どれ as I mouse over it in different directions.

![output](https://user-images.githubusercontent.com/11037431/73126138-18566b00-3f7d-11ea-8f7d-2772739fa958.gif)

I also tested this in Firefox and Firefox does not have the issue. Firefox's text seems to render slightly differently also, but there is a clear difference between ClearType enabled vs disabled. So I'm not sure what Chrome is doing differently.

**Comment by sailikk** [comment]
> There's an additional orange highlight on the selected text in your image. Do you know what that is coming from?

This is coming from the Ctrl-F find function on Chrome, I was looking for それなら, but I didn't close it off before hovering over the text haha.

I never messed with my ClearType font rendering, so it was on my default it seems. 

> I had to use the "Advanced scaling settings" window to get a scaling of 120%; did you also have to use that window?

Yes, exactly.

**Comment by toasted-nutbread** [comment]
Okay, I think I have determined what the bug is. It seems to be related to the scrollbar's presence. I've created a repository demonstrating the bug and filed a Chrome bug report.

https://toasted-nutbread.github.io/chrome-layout-bug/
https://github.com/toasted-nutbread/chrome-layout-bug
https://bugs.chromium.org/p/chromium/issues/detail?id=1045755

The good news is that I tested this issue using the latest development branch and the issue is fixed, likely because the scrollbar is persistent on the popup.
https://github.com/FooSoft/yomichan/commit/97f7df33027e8c8baced23da02b74a5fbd0a14cf#diff-604777a6212329d0b021db678429e058R45

 So when a new release gets pushed, this issue should be fixed. I'll close the issue once that happens.

**Comment by siikamiika** [comment]
Nice job! The font is the regular Meiryo, not Meiryo UI, right?
![](https://upload.wikimedia.org/wikipedia/commons/4/4f/Meiryo_VS_Meiryo_UI.svg)
As seen above, in Meiryo UI kana is not as wide as kanji.

**Comment by toasted-nutbread** [comment]
I get the same issue with Meiryo UI, but the default font is just Meiryo. The issue can also occur when forcing the tab's scaling level to 120% while the system DPI scaling is at 100%. There's not really a good way to do this since 120% is not a default scaling option, but it can be forced from a Yomichan tab's console:

```js
chrome.tabs.query({
  url: '<whatever other tab URL>'
}, (tabs) => {
  chrome.tabs.setZoom(tabs[0].id, 1.2, () => { console.log('set zoom'); }) 
}) 
```

**Comment by siikamiika** [comment]
Reproduced on Chromium 79.0.3945.79 on Linux with Noto Sans CJK JP:
![2020-01-26-045233_1252x649_scrot](https://user-images.githubusercontent.com/5602483/73129958-1251a000-3ff8-11ea-9ccf-8207f397206d.png)

**Comment by toasted-nutbread** [comment]
Interestingly, I have encountered this bug myself after I thought that it was fixed. Slightly different context for it occurring, but the effect is very similar. Still an issue in Chrome. The popup can't be resized this way in Firefox, but some manual resizing doesn't seem to cause the issue. Zoom level at 110% this time, a standard zoom level available on Chrome.

![test](https://user-images.githubusercontent.com/11037431/74206945-7a84c080-4c4b-11ea-89f2-4cae700d8288.gif)
